lib/cli/main.go

```diff
@@ -15,23 +15,30 @@
 package cli
 
 import (
+	"crypto/tls"
 	"net/http"
 
+	"github.com/pingcap/TiProxy/lib/config"
+	"github.com/pingcap/TiProxy/lib/util/security"
 	"github.com/spf13/cobra"
 	"go.uber.org/zap"
 )
 
-func GetRootCmd() *cobra.Command {
+func GetRootCmd(tlsConfig *tls.Config) *cobra.Command {
 	rootCmd := &cobra.Command{
-		Use:   "tiproxyctl",
-		Short: "cli",
+		Use:          "tiproxyctl",
+		Short:        "cli",
+		SilenceUsage: true,
 	}
 
 	ctx := &Context{}
 
 	curls := rootCmd.PersistentFlags().StringArray("curls", []string{"localhost:3080"}, "API gateway addresses")
 	logEncoder := rootCmd.PersistentFlags().String("log_encoder", "tidb", "log in format of tidb, console, or json")
 	logLevel := rootCmd.PersistentFlags().String("log_level", "info", "log level")
+	caPath := rootCmd.PersistentFlags().String("ca", "", "CA to verify server certificates, set to 'skip' if want to enable SSL without verification")
```

Comment:
Why need to set `skip`? Why not set it to an empty string or skip setting it?

Reply:
1. `""` means no TLS, retry will use the system CA
2. `skip` means TLS without CA, fallback to HTTP if server may be HTTP
3. otherwise, it is normal TLS with specified CA, fallback to HTTP if server may be HTTP

So, in short, `skip` is like `curl -k`, give a chance to use TLS insecurely(for example testing).

Reply:
You have `certPath` and `keyPath`. If `certPath` and `keyPath` are specified, then TLS is enabled without verification. This is what TiDB does in the SQL port.

Reply:
That is not how tiproxy works. I invoked `BuildClientTLSConfig`, only CA or skip-ca will enable tls. Certs are ignored without ca.